### PR TITLE
gh-110696: Fix incorrect syntax error message for incorrect argument unpacking

### PR DESCRIPTION
**Pull request by pablogsal**
* Issue: gh-110696

### CODE REVIEW

File: Grammar/python.gram [hunk 1]
```diff
@@ -1128,7 +1128,10 @@ func_type_comment[Token*]:
 
 # From here on, there are rules for invalid syntax with specialised error messages
 invalid_arguments:
-    | a=args ',' '*' { RAISE_SYNTAX_ERROR_KNOWN_LOCATION(a, "iterable argument unpacking follows keyword argument unpacking") }
+
+
```
Comment (lysnikolaou) [inline Grammar/python.gram]: Remove the two empty lines?

**Review by lysnikolaou — COMMENTED** [review]
One nit.

File: Grammar/python.gram [hunk 2]
```diff
@@ -1128,7 +1128,10 @@ func_type_comment[Token*]:
...
+    | ((','.(starred_expression | ( assignment_expression | expression !':=') !'=')+ ',' kwargs) | kwargs) ',' b='*' {
+        RAISE_SYNTAX_ERROR_KNOWN_LOCATION(b, "iterable argument unpacking follows keyword argument unpacking") }
```
Comment (pablogsal) [inline Grammar/python.gram]:
```suggestion
    | ((','.(starred_expression | ( assignment_expression | expression !':=') !'=')+ ',' kwargs) | kwargs) ',' b='*' {
        RAISE_SYNTAX_ERROR_KNOWN_LOCATION(b, "iterable argument unpacking follows keyword argument unpacking") }
```

**Review by lysnikolaou — APPROVED** [review]
LGTM!

**Comment by lysnikolaou** [comment]
Tests are failing still.

**Pull request merged**